File: Graph/BFS.py
#Breadth First Search

class Adjacent_list:
	def __init__(self,n):
		self.lis=[[] for _ in range(n)]

class Queue:
	def __init__(self):
		self.queue=[]

	def enqueue(self,v):
		self.queue.append(v)

	def dequeue(self):
		x = self.queue.pop(0)
		return x

	def is_Empty(self):
		return len(self.queue) == 0



class BFS(Adjacent_list):
	
	def add_edges(self,v1,v2):
		self.lis[v1].append(v2)
		self.lis[v2].append(v1)

	def bfs(self,source):
		visited=[]
		Q=Queue()
		Q.enqueue(source)
		visited.append(source)
		print(source,end=" ")
		while not Q.is_Empty():
			node=Q.dequeue()
			for neighbour in self.lis[node]:
				if neighbour not in visited:
					visited.append(neighbour)
					print(neighbour,end=" ")
					Q.enqueue(neighbour)


def main():
	
	n=int(input("Enter the number of vertices:"))
	B=BFS(n)
	#creating an object for both the classes

	print ("Enter the edges(x to stop):")
	while True:
		e=input().split()
		if e[0]=='x':
			break
		else:
			v1,v2=map(int,e)
			B.add_edges(v1,v2)

	print (B.lis)
			
	print("Enter the source vertex:")
	source=int(input())
	B.bfs(source)

if __name__ == '__main__':
	main()

File: Graph/BFS.cpp
#include<bits/stdc++.h>
using namespace std;
class Graph
{
   int v;
   list<int> *adjlist;

   public:
   Graph(int v);
   void addedges(int v1,int v2);
   void BFS(int src);
};

Graph::Graph(int v){
this->v=v;
adjlist=new list<int>[v];
}

void Graph::addedges(int v1,int v2){
adjlist[v1].push_back(v2);
}


void Graph::BFS(int src){
bool *visited = new bool[v];
for(int i=0;i<v;i++){
    visited[i]=false;
}
queue<int>q;
visited[src]=true;
q.push(src);
while(!q.empty()){
    int v=q.front();
    cout<<v<<" ";
    q.pop();
    list<int>::iterator i;
    for(i=adjlist[v].begin();i!=adjlist[v].end();i++){
        if(visited[*i]==false){
            q.push(*i);
            visited[*i]=true;
        }
    }
}
}

int main(){
Graph g(4);
    g.addedges(0, 1);
    g.addedges(0, 2);
    g.addedges(1, 2);
    g.addedges(2, 0);
    g.addedges(2, 3);
    g.addedges(3, 3);

    cout << "Following is Breadth First Traversal "
         << "(starting from vertex 2) \n";
    g.BFS(2);

    return 0;
}


